libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Generators/ActivityChecker.cs

```diff
@@ -0,0 +1,372 @@
+
+                            break;
+                        default:
+                            result.Add(BuildDiagnostic($"'{property}' is not support for card action.", false));
+                            break;
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        private static string GetStructureType(JObject jObj)
+        {
+            if (jObj == null)
+            {
+                return string.Empty;
+            }
+
+            var type = jObj["$type"]?.ToString()?.Trim();
+            if (string.IsNullOrEmpty(type))
+            {
+                // Adaptive card type
+                type = jObj["type"]?.ToString()?.Trim();
+            }
+
+            return type?.ToLower() ?? string.Empty;
+        }
+
+        private static List<Diagnostic> CheckAttachments(JToken value)
+        {
+            var result = new List<Diagnostic>();
+
+            var attachmentsJsonList = NormalizedToList(value);
+
+            foreach (var attachmentsJson in attachmentsJsonList)
+            {
+                if (attachmentsJson is JObject attachmentsJsonJObj)
+                {
+                    result.AddRange(CheckAttachment(attachmentsJsonJObj));
+                }
+            }
+
+            return result;
+        }
+
+        private static List<Diagnostic> CheckAttachment(JObject lgJObj)
+        {
+            var result = new List<Diagnostic>();
+
+            var type = GetStructureType(lgJObj);
+
+            if (ActivityFactory.GenericCardTypeMapping.ContainsKey(type))
+            {
+                result.AddRange(CheckCardAtttachment(ActivityFactory.GenericCardTypeMapping[type], lgJObj));
+            }
+            else if (type == nameof(AdaptiveCard).ToLower())
+            {
+                try
+                {
+                    var parseResult = AdaptiveCard.FromJson(lgJObj.ToString());
+                    if (parseResult.Warnings.Count > 0)
+                    {
+                        foreach (var warning in parseResult.Warnings)
+                        {
+                            result.Add(BuildDiagnostic(warning.Message, false));
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    result.Add(BuildDiagnostic(e.Message));
+                }
+            }
+            else
+            {
+                result.Add(BuildDiagnostic($"'{type}' is not an attachment type.", false));
+            }
+
+            return result;
+        }
+
+        private static List<Diagnostic> CheckCardAtttachment(string type, JObject lgJObj)
+        {
+            var result = new List<Diagnostic>();
+
+            foreach (var item in lgJObj)
+            {
+                var property = item.Key.Trim().ToLower();
+                var value = item.Value;
+
+                switch (property)
+                {
+                    case "$type":
```

Comment:
This list seems fragile. Is there a better and more maintainable way to code this?